pkg/controllers/machine/lifecycle/launch.go

```diff
@@ -107,6 +109,13 @@ func (l *Launch) launchMachine(ctx context.Context, machine *v1alpha5.Machine) (
 			machine.StatusConditions().MarkFalse(v1alpha5.MachineLaunched, "LaunchFailed", truncateMessage(err.Error()))
 			return nil, fmt.Errorf("creating machine, %w", err)
 		}
+		l.recorder.Publish(events.Event{
```

Comment:
Should we convert this flow into a switch case on the error type and then default to the error return if we don't recognize it? That structure would make adding more error-specific events in the future easier.

Reply:
Yeah, I think that flow would be a lot easier to reason about. It took me a second to catch that the if was `Not an InsufficientCapacityError`

Reply:
Fixed :) I added a `switch` to check the error type.